core/src/player.rs

```diff
@@ -1252,15 +1252,54 @@ impl Player {
                 ));
             }
 
+            let mut new_over_object_updated = false;
             // Cancel hover if an object is removed from the stage.
             if let Some(hovered) = context.mouse_over_object {
                 if !context.is_action_script_3() && hovered.as_displayobject().avm1_removed() {
                     context.mouse_over_object = None;
+                    if let Some(new_object) = new_over_object {
+                        if new_object.as_displayobject().id() == hovered.as_displayobject().id()
+                            && new_object.as_displayobject().depth()
+                                == hovered.as_displayobject().depth()
+                            && new_object.as_displayobject().movie().url()
+                                == hovered.as_displayobject().movie().url()
```

Comment:
You'll want to use pointer equality instead of checking the URL, since it's possible there's multiple different movies loaded from the same URL. The syntax for actually checking that is `Arc::ptr_eq(&new_object.as_displayobject().movie(), &hovered.as_displayobject().movie())`.